enhancements/authentication/oauth-resource-storage.md

```diff
@@ -0,0 +1,253 @@
+e.g. in order to show that information via oc or console, or to logout certain tokens.
+
+### Goals
+
+- Change the `oauthaccesstokens` and `oauthauthorizetokens` storage format and schema such that
+  the etcd keys alone (which are not encrypted when encryption at REST is enabled) are not sensitive.
+- Allow console and oc to log out all other sessions.
+
+### Non-Goals
+
+## Proposal
+
+We extend the OAuthAccessToken and OAuthAuthorizeToken types in the following way:
+
+```go
+// OAuthAccessToken describes an OAuth access token
+type OAuthAccessToken struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    // ClientName references the client that created this token.
+    ClientName string `json:"clientName,omitempty" protobuf:"bytes,2,opt,name=clientName"`
+
+    // ExpiresIn is the seconds from CreationTime before this token expires.
+    ExpiresIn int64 `json:"expiresIn,omitempty" protobuf:"varint,3,opt,name=expiresIn"`
+
+    // Scopes is an array of the requested scopes.
+    Scopes []string `json:"scopes,omitempty" protobuf:"bytes,4,rep,name=scopes"`
+
+    // RedirectURI is the redirection associated with the token.
+    RedirectURI string `json:"redirectURI,omitempty" protobuf:"bytes,5,opt,name=redirectURI"`
+
+    // UserName is the user name associated with this token
+    UserName string `json:"userName,omitempty" protobuf:"bytes,6,opt,name=userName"`
+
+    // UserUID is the unique UID associated with this token
+    UserUID string `json:"userUID,omitempty" protobuf:"bytes,7,opt,name=userUID"`
+
+    // AuthorizeToken contains the token that authorized this token
+    // +sensitive
+    AuthorizeToken string `json:"authorizeToken,omitempty" protobuf:"bytes,8,opt,name=authorizeToken"`
+
+    // RefreshToken is the value by which this token can be renewed. Can be blank.
+    // +sensitive
+    RefreshToken string `json:"refreshToken,omitempty" protobuf:"bytes,9,opt,name=refreshToken"`
+
+    // InactivityTimeoutSeconds is the value in seconds, from the
+    // CreationTimestamp, after which this token can no longer be used.
+    // The value is automatically incremented when the token is used.
+    InactivityTimeoutSeconds int32 `json:"inactivityTimeoutSeconds,omitempty" protobuf:"varint,10,opt,name=inactivityTimeoutSeconds"`
+
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAccessToken object.
+    // For old tokens, this field might be empty.
+    // +sensitive
+    // +optional
+    SecretToken string `json:"secretToken,omitempty" protobuf:"bytes,11,opt,name=secretToken"`
+}
+
+// OAuthAuthorizeToken describes an OAuth authorization token
+type OAuthAuthorizeToken struct {
+    metav1.TypeMeta   `json:",inline"`
```

Comment:
how about removing already existing fields to show only a diff?

Reply:
done